client/src/utils/fileLock.ts

```diff
@@ -0,0 +1,70 @@
+import {open, unlink, readFile} from 'fs/promises';
+
+export class FileLockError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = 'FileLockError';
+  }
+}
+export async function acquireLock(lockFilePath: string): Promise<boolean> {
+  try {
+    const fileHandle = await open(lockFilePath, 'wx');
+    await fileHandle.writeFile(String(process.pid));
+    await fileHandle.close();
+    return true;
+  } catch (error) {
+    const err = error as NodeJS.ErrnoException;
+    if (err.code === 'EEXIST') {
+      return await handleExistingLock(lockFilePath);
+    }
+    throw new FileLockError(`Failed to acquire lock: ${(error as Error).message}`);
+  }
+}
+
+export async function releaseLock(lockFilePath: string): Promise<void> {
+  try {
+    await unlink(lockFilePath);
+  } catch (error) {
+    const err = error as NodeJS.ErrnoException;
+    if (err.code === 'ENOENT') {
+      return;
+    }
+    throw new FileLockError(`Failed to release lock: ${(error as Error).message}`);
+  }
+}
+
+async function handleExistingLock(lockFilePath: string): Promise<boolean> {
+  try {
+    const pidInLock = await readFile(lockFilePath, 'utf-8');
+    process.kill(Number(pidInLock), 0);
+    return false;
+  } catch (error) {
+    const err = error as NodeJS.ErrnoException;
+    if (err.code === 'ESRCH') {
+      await releaseLock(lockFilePath);
+      return await acquireLock(lockFilePath);
```

Comment:
락 파일 생성/삭제가 빠르게 반복되면 무한 재귀가 발생할 위험이 있을 것 같아요!

1. 락 파일이 존재하지만
2. 해당 프로세스가 죽어있어서(좀비 프로세스)
3. 락 파일을 삭제하고
4. 다시 acquireLock 호출(재귀)

좀비 프로세스 정리 후에 바로 상위에서 재시도하는 방식으로 개선해보는건 어떨까요?

```suggestion
      return false; // 재귀 대신 false 반환하여 상위에서 재시도
```

Reply:
추가로 이렇게 반환값을 세분화할 수도 있을것 같네요!

```
  enum LockResult {
    SUCCESS = 'success',           // 락 획득 성공
    BUSY = 'busy',                 // 다른 프로세스가 사용 중
    ZOMBIE_CLEANED = 'zombie',      // 좀비 프로세스 정리함
    PERMISSION_ERROR = 'permission', // 파일 권한 문제
    UNKNOWN_ERROR = 'error'        // 기타 에러
  }
```